keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -291,113 +697,285 @@ No, the default behavior remains the same.
+
+1. Create a cluster in 1.33.
+2. Upgrade to 1.34.
+3. Create StatefulSet A with spec.updateStrategy.rollingUpdate.maxUnavailable set to 3, with 6 replicas
+4. Verify a rollout and check if only 3 pods are unavailable at a time ([currently with a bug if podManagementPolicy is set to Parallel](https://github.com/kubernetes/kubernetes/issues/112307))
+5. Downgrade to 1.33.
+6. Verify that the rollout only has 1 pod unavailable at a time, similar to setting maxUnavailable to 1
+7. Create another StatefulSet B not setting maxUnavailable (leaving it nil)
+8. Upgrade to 1.34.
+9. Verify that the rollout has default behavior of only having one pod unavailable at a time
+   Verify that the `maxUnavailable` can be set again to StatefulSet A and test the rollout behavior
+
+TODO:
+A manual test will be performed, as follows:
+
+1. Create a cluster in 1.33.
+2. Upgrade to 1.34.
+3. Create StatefulSet A with spec.updateStrategy.rollingUpdate.maxUnavailable set to 3, with 6 replicas
+4. Verify a rollout and check if only 3 pods are unavailable at a time
+5. Check if rollout is also fine with podManagementPolicy set to Parallel
+6. Downgrade to 1.33.
+7. Verify that the rollout only has 1 pod unavailable at a time, similar to setting maxUnavailable to 1 (MaxUnavailableStatefulSet feature gate disabled by default).
+8. Create another StatefulSet B not setting maxUnavailable (leaving it nil)
+9. Upgrade to 1.34.
+10. Verify that the rollout has default behavior of only having one pod unavailable at a time
+    Verify that the `maxUnavailable` can be set again to StatefulSet A and test the rollout behavior
 
 ###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
 No
 
 ### Monitoring Requirements
 
 ###### How can an operator determine if the feature is in use by workloads?
+
 If their StatefulSet rollingUpdate section has the field maxUnavailable specified with
-a value different than 1.
-The below command should show maxUnavailable value:
+a value different than 1. While in alpha and beta, the feature-gate needs to be enabled.
+
+The command bellow should show the maxUnavailable value:
+
 ```
 kubectl get statefulsets -o yaml | grep maxUnavailable
 ```
 
 ###### How can someone using this feature know that it is working for their instance?
-TODO when we reach Beta
+
+Users can verify the maxUnavailable feature is working for their StatefulSets by observing the behavior of their application during rolling updates. Specifically, they should monitor the update progress of StatefulSets against the maxUnavailable value they have configured. This can be done by tracking the number of pods that are unavailable at any given time during the update process and ensuring it does not exceed the specified maxUnavailable limit. Users can also view the `statefulset_unavailability_violation` metric to see if there have been instances
+where the feature is not working as intended.
```

Comment:
You want to follow the [template](https://github.com/kubernetes/enhancements/blob/master/keps/NNNN-kep-template/README.md#how-can-an-operator-determine-if-the-feature-is-in-use-by-workloads) here and add something like:
```
- [ ] Events
  - Event Reason: 
- [x] API .spec
  - Condition name: 
  - Other field: .spec.updateStrategy.rollingUpdate.maxUnavailable
- [ ] Other (treat as last resort)
  - Details:
 ```